src/methods/abstract.js

```diff
@@ -74,6 +75,26 @@ export function RequireObjectCoercible(
   argLoc?: ?BabelNodeSourceLocation
 ): AbstractValue | ObjectValue | BooleanValue | StringValue | SymbolValue | NumberValue {
   if (arg instanceof AbstractValue && (arg.mightBeNull() || arg.mightBeUndefined())) {
+    if (realm.isInPureScope()) {
+      if (realm.isInPureTryStatement) {
+        // TODO(1264): We should be able to preserve error handling on abstract throw
+        // but currently we just issue a recoverable error instead.
+        let diag = new CompilerDiagnostic(
```

Comment:
Please add a test where this diagnostic is actually produced.

Reply:
I think that I'll actually remove this one. I wanted to ensure that we always have a warning but this will be covered in other places anyway such as in $Get in this case so if you recover it'll be issued twice which is unnecessary.

We also need any solution to #1264 to be colocated with the op that actually forces this so it doesn't make sense here.

I'll add a test for the other one though.